### DISCUSSION TITLE
Public name conflict method and property

### DISCUSSION THREAD

**Discussion by schenke-io** [livewire/livewire#7135] [7. Bugs (archived)]
### Livewire version

v3.0.10

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

create a livewire class
give it a method   `public function myConflictName() { }`
add a property with same name `public string myConflictName = '';`
in the blade add somehow a `wire:click="myConflictName" `
no click event happens

```
<div>
    <button wire:click="myConflictName">
        Property: {{$myConflictName}}
    </button>
</div>

```

```
<?php

namespace App\Livewire;

use Livewire\Component;

class BugDemo extends Component
{
    public string $myConflictName = '';

    public function myConflictName()
    {
        $this->myConflictName = "test";
    }

    public function render()
    {
        return view('livewire.bug-demo');
    }
}

```

### Additional information

Maybe the bug happens seldom since properties and methods are never named the same.
But it would be good to get an error message

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by andreas-freitag** [comment]
Even though the change has lead to a lot of work in my project, it is mentioned in the upgrade guide (nearly at the bottom of the page):

[Upgrading](https://livewire.laravel.com/docs/upgrading)
